### PR TITLE
[Reader] Remove improvements FF and unused UI files

### LINKED ISSUE #19261: [Reader Improvements] Remove all the Reader files from the previous layout
For the `Reader Improvements` feature we've been duplicating many files to simplify keeping the changes behind a feature flag. The new files usually have a `_new` suffix. After the feature is live and 🟢 , we should remove the `Reader Improvements` feature flag and also the "old" layout files not being used anymore.

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #19261 

-----

## To Test:

Basically the Reader should continue to work as expected, including all the feeds (`Discover`, `Subscriptions`, `Saved`, `Liked`, `Your Tags` and custom lists), post details, recommendation cards on `Discover`, actions, etc.
It would be nice to also test it out using light and dark themes since the PR is related to UI.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Reader post lists and post details.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Updated `ReaderDiscoverViewModelTest.kt`.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ChipStyle</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt [hunk 1]
```diff
@@ -73,47 +73,6 @@ class ReaderPostUiStateBuilder @Inject constructor(
...
-                onVideoOverlayClicked,
-                onPostHeaderViewClicked,
-                onTagItemClicked,
-                moreMenuItems
-            )
-        }
-    }
-
     @Suppress("LongParameterList")
     fun mapPostToUiStateBlocking(
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt]: It looks like this is still being used somewhere, is that correct? Doesn't it map to the old UI State for the old post UI?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderExpandableTagsView.kt [hunk 2]
```diff
@@ -200,34 +196,12 @@ class ReaderExpandableTagsView @JvmOverloads constructor(
         @ColorRes
         fun overflowStrokeColorRes(isCollapsed: Boolean): Int? = null
 
-        object Legacy : ChipStyle {
+        data object Default : ChipStyle {
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderExpandableTagsView.kt]: Now that we have just one implementation, what do you think about making `ChipStyle` a concrete class and moving the implementation from this subclass to there?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderPostDetailHeaderView.kt [hunk 3]
```diff
@@ -153,45 +138,8 @@ class ReaderPostDetailHeaderView @JvmOverloads constructor(
...
-
-            override fun updateInteractionSection(
-                state: InteractionSectionUiState,
-                readingPreferences: ReaderReadingPreferences?
-            ) {
-                // do nothing
-            }
-        }
-
         class ImprovementsEnabled(
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderPostDetailHeaderView.kt]: Now that we have just one implementation, what do you think about making `ReaderPostDetailHeaderBinding` a concrete class and moving the implementation from this subclass to there?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.kt [hunk 4]
```diff
@@ -28,32 +26,21 @@ class ReaderTagHeaderView @JvmOverloads constructor(
...
 
     init {
         (context.applicationContext as WordPress).component().inject(this)
-        binding = if (readerImprovementsFeatureConfig.isEnabled()) {
-            val readerTagHeaderViewNewBinding =
-                ReaderTagHeaderViewNewBinding.inflate(LayoutInflater.from(context), this, true)
+        val readerTagHeaderViewBinding =
+            ReaderTagHeaderViewBinding.inflate(LayoutInflater.from(context), this, true)
+        binding =
             ReaderTagBinding.ImprovementsEnabled(
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.kt]: Now that we have just one implementation, what do you think about making `ReaderTagBinding` a concrete class and moving the implementation from this subclass to there?

Also, delete the now unused `ImprovementsDisabled` class (which will be necessary anyway if you implement the suggestion above).

**Review by thomashorta — COMMENTED** [review]
Overall the code looks good but I noticed a few places that the `New` nomenclature is still being used (in some class names and Theme styles) and also found a place where we are mapping to a "legacy" UI model and I'm not sure if that means we're still showing the old post UI somewhere, can you double check?

**Comment by RenanLukas** [comment]
Thanks for the review, @thomashorta !
I've implemented your suggestions and if CI is 🟢  it should be ready for a second review.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt]: For some reason I thought it was being used, but it seems like it was just being used in the unit tests. Code updated.

**Review by thomashorta — APPROVED** [review]

**Review by nbradbury — APPROVED** [review]
As discussed elsewhere, we've decided to close older PRs that have non-trivial merge conflicts  and/or have non-trivial CI issues. Based on that, this should be closed. However, given that this removes 3001 lines of outdated code and it was previously approved, I felt it worthwhile to update it and get it working. :shipit:

**Pull request merged**